### PR TITLE
Graph scrolling forever

### PR DESCRIPTION
**Pull request by Vane11ope**
Graph now scrolls forever and Overview also applies for it.

even when only a node is shown on the screen, you can move the node.

Also refactoring a lot of Graph and Overview code.

gifrecord_2019-02-14_202137
gifrecord_2019-02-14_201737

### CODE REVIEW

**Comment by ITAYC0HEN** [comment]
Can't check until agJ would be fixed

File: src/widgets/GraphView.h [hunk 1]
```diff
@@ -95,15 +95,14 @@ class GraphView : public QAbstractScrollArea
...
-    void showBlock(GraphBlock *block, bool animated = false);
+    void showBlock(GraphBlock &block);
+    void showBlock(GraphBlock *block);
 
     // Zoom data
     qreal current_scale = 1.0;
 
-    int unscrolled_render_offset_x = 0;
-    int unscrolled_render_offset_y = 0;
+    int offset_x = 0;
```
Comment (xarkes) [inline src/widgets/GraphView.h]: I would add a comment to specify what's the meaning of `offset_x` and `offset_y`.
Because I know what your PR does, I know what the offset is for, but otherwise it can be tricky to understand imho.
I believe a `\brief` would be helpful here.

File: src/widgets/GraphView.cpp [hunk 2]
```diff
@@ -727,46 +675,43 @@ int GraphView::findVertEdgeIndex(EdgesVector &edges, int col, int min_row, int m
...
-    target_y = std::max(0, target_y);
-    target_x = std::min(horizontalScrollBar()->maximum(), target_x);
-    target_y = std::min(verticalScrollBar()->maximum(), target_y);
-    if (animated) {
-        QPropertyAnimation *animation_x = new QPropertyAnimation(horizontalScrollBar(), "value", this);
-        animation_x->setDuration(500);
-        animation_x->setStartValue(horizontalScrollBar()->value());
-        animation_x->setEndValue(target_x);
-        animation_x->setEasingCurve(QEasingCurve::InOutQuad);
-        animation_x->start(QAbstractAnimation::DeleteWhenStopped);
```
Comment (xarkes) [inline src/widgets/GraphView.cpp]: Why did you remove this? Is it handled by `centerX` followed with `blockTransitionedTo`?

File: src/widgets/GraphView.h [hunk 3]
```diff
@@ -95,15 +95,14 @@ class GraphView : public QAbstractScrollArea
     ~GraphView() override;
     void paintEvent(QPaintEvent *event) override;
 
-    // Show a block centered. Animates to it if animated=true
-    void showBlock(GraphBlock &block, bool animated = false);
-    void showBlock(GraphBlock *block, bool animated = false);
```
Comment (xarkes) [inline src/widgets/GraphView.h]: So everything is animated now?

File: src/widgets/GraphView.h (hunk 1 above)
Comment (Vane11ope) [reply src/widgets/GraphView.h]: commenting comes to the whole graph code after this gets merged, not only here.

File: src/widgets/GraphView.cpp (hunk 2 above)
Comment (Vane11ope) [reply src/widgets/GraphView.cpp]: because I just thought animation only slows down the users and does not mean anything. this was just copy-pasted from x64dbg code anyway and i don't think we wanna keep this.

File: src/widgets/GraphView.h (hunk 3 above)
Comment (Vane11ope) [reply src/widgets/GraphView.h]: no, intentionally.

File: src/widgets/GraphView.cpp (hunk 2 above)
Comment (xarkes) [reply src/widgets/GraphView.cpp]: No, the animation was added after x64dbg and does not come from their codebase.
If they have it nowadays it's probably because they took it from us.

**Comment by ITAYC0HEN** [comment]
Thank you! Very nice :D
